cypress/e2e/WidgetsForEnrollmentPages/WidgetTeiRelationship/index.js

```diff
@@ -0,0 +1,146 @@
+                                    updatedAt: '2024-02-01T11:50:25.479',
+                                    valueType: 'TEXT',
+                                    value: 'John',
+                                },
+                                {
+                                    attribute: 'lZGmxYbs97q',
+                                    code: 'MMD_PER_ID',
+                                    displayName: 'Unique ID',
+                                    createdAt: '2024-02-01T11:50:25.476',
+                                    updatedAt: '2024-02-01T11:50:25.476',
+                                    valueType: 'TEXT',
+                                    value: '0078200',
+                                },
+                                {
+                                    attribute: 'zDhUuAYrxNC',
+                                    displayName: 'Last name',
+                                    createdAt: '2024-02-01T11:50:25.479',
+                                    updatedAt: '2024-02-01T11:50:25.479',
+                                    valueType: 'TEXT',
+                                    value: 'Mayer',
+                                },
+                            ],
+                        },
+                    },
+                },
+            ],
+        },
+    }).as('getRelationships');
+});
+
+Given('the user can see the relationship widget', () => {
+    cy.get('[data-test="tracked-entity-relationship-widget"]')
+        .should('be.visible');
+});
+
+When('there is an existing relationship', () => {
+    cy.wait('@getRelationships');
+    cy.get('[data-test="tracked-entity-relationship-widget"]')
+        .within(() => {
+            cy.get('[data-test="relationship-table-row"]')
+                .contains('John');
+        });
+});
+
+When('the user clicks the delete button', () => {
+    cy.get('[data-test="tracked-entity-relationship-widget"]')
+        .within(() => {
+            cy.get('[data-test="relationship-table-row"]')
+                .contains('John')
+                .parent()
+                .within(() => {
+                    cy.get('[data-test="delete-relationship-button"]')
+                        .click();
+                });
+        });
+});
+
+When('the user can see the delete relationship modal', () => {
+    cy.get('[data-test="delete-relationship-modal"]').should('be.visible');
+});
+
+When('the user clicks the confirm delete button', () => {
+    cy.intercept({
+        method: 'POST',
+        url: '**/tracker?importStrategy=DELETE&async=false**',
+    }).as('deleteRelationship');
+
+    cy.get('[data-test="delete-relationship-modal"]')
+        .within(() => {
+            cy.get('[data-test="delete-relationship-confirmation-button"]')
+                .click();
+        });
+
+    cy.wait('@deleteRelationship')
+        .its('request.body')
+        .should('deep.equal', {
+            relationships: [{ relationship: 'mwswG3RMNuu' }],
```

Comment:
**Reminder** after the bug in the API is fixed: To keep DB consistency, the relationship that was deleted should be added back. Otherwise, the `there is an existing relationship` step will fail the next time the scenario is run.

Reply:
This data is actually just mocked, so no need to add it again. We intercept the request for relationships in the before hook and then assert that we send the actual call to the api with the correct data! WDYT?

Reply:
Testing against mocked data versus the API and DB has advantages and disadvantages 🤔 . By using mock data, we will no longer benefit from having the latest API changes available. We will need to ensure that the mocks are always up to date, otherwise, we may not detect when we introduce bugs. 
Let's discuss it more in-depth in our next FE check-in meeting!